src/test/store/profile-view.test.js

```diff
@@ -2443,3 +2456,187 @@ describe('visual metrics selectors', function() {
+      .fill(0)
+      .map((_, i) => i);
+    profile.threads.push(getEmptyThread({ samples }));
+
+    const { dispatch, getState } = storeWithProfile(profile);
+
+    const getProcessedEventDelays = () =>
+      profile.threads.map((_, threadIndex) =>
+        getThreadSelectors(threadIndex).getProcessedEventDelays(getState())
+      );
+
+    return { profile, dispatch, getState, getProcessedEventDelays };
+  }
+
+  it('can process the event delay values and returns meaningful numbers', function() {
+    const { getProcessedEventDelays } = setup();
+    expect(getProcessedEventDelays()).toEqual([
+      {
+        maxDelay: 52,
+        minDelay: 1,
+        delayRange: 51,
+        eventDelays: [
+          0,
+          1,
+          1,
+          1,
+          1,
+          52, // <---- Event delay peak.
+          51,
+          50,
+          49,
+          48,
+          47,
+          46,
+          45,
+          44,
+          43,
+          42,
+          41,
+          40,
+          39,
+          38,
+          37,
+          36,
+          35,
+          34,
+          33,
+          32,
+          31,
+          30,
+          29,
+          28,
+          27,
+          26,
+          25,
+          24,
+          23,
+          22,
+          21,
+          20,
+          19,
+          18,
+          17,
+          16,
+          15,
+          14,
+          13,
+          12,
+          11,
+          10,
+          9,
+          7,
+          6,
+          5,
+          4,
+          3,
+          2,
+          1, // <---- goes down until it's done.
+        ],
+      },
+    ]);
+  });
+
+  it('can process the event delay values with two combined peaks and returns meaningful numbers', function() {
+    const eventDelay = Array(50).fill(0);
+    eventDelay.push(10, 20, 30, 40, 50, 0, 0, 10, 20, 0);
+    //                              ^             ^
+    //                              |             |
+    //                              First peak    Second peak
+    const { getProcessedEventDelays } = setup(eventDelay);
+    expect(getProcessedEventDelays()).toEqual([
+      {
+        maxDelay: 52,
+        minDelay: 1,
+        delayRange: 51,
+        eventDelays: [
+          0,
+          1,
+          1,
+          1,
+          1,
+          52, // <---- First event delay peak.
+          51,
+          50,
+          49,
+          48,
+          47,
+          46,
+          45,
+          44,
+          43,
+          42,
+          41,
+          40,
+          39,
+          38,
+          37,
+          36,
+          35,
+          34,
+          33,
+          32,
+          31,
+          30,
+          29,
+          28,
+          27,
+          26,
+          25,
+          24,
+          23,
+          22,
+          21,
+          20,
+          19,
+          39, // <---- Second event delay peak. This happens while we are still no done with the first even delay and sums up the both delay values.
```

Comment:
```suggestion
          39, // <---- Second event delay peak. This happens while we are still not done with the first event delay and sums up both delay values.
```

Also maybe put the comment on the line before, and make it a multiline comment.